yarn-project/end-to-end/src/guides/writing_an_account_contract.test.ts

```diff
@@ -59,14 +66,20 @@ describe('guides/writing_an_account_contract', () => {
     logger(`Deployed account contract at ${address}`);
 
     // docs:start:account-contract-works
-    const token = await PrivateTokenContract.deploy(wallet, 100, address).send().deployed();
+    const token = await TokenContract.deploy(wallet).send().deployed();
     logger(`Deployed token contract at ${token.address}`);
+    await token.methods._initialize({ address }).send().wait();
 
-    await token.methods.mint(50, address).send().wait();
-    const balance = await token.methods.getBalance(address).view();
+    const secret = Fr.random();
+    const secretHash = await computeMessageSecretHash(secret);
+
+    await token.methods.mint_private(50, secretHash).send().wait();
```

Comment:
in general all of these could probably be sped up in the tests by sending the promises together and doing a `Promise.all`. 
Or is this not doable as we require the mint private to happen before the redeem

Reply:
If done with promise all it still got in as separate txs, so it cannot perform the second part that depends on the first. Might be replaces with requests I guess to do it in the same tx.

Reply:
Nvm, we can't do it in same tx because the mint_private is a public call and we need the result in private.